workflows/chatbot/fine_tuning/dpo_pipeline/utils.py

```diff
@@ -0,0 +1,118 @@
+from collections import defaultdict
```

Comment:
should this file be under llm?

dpo_trainer.py should also be under llm?

Reply:
dpo_trainer.py is under llm